main.go

```diff
@@ -65,13 +68,20 @@ func main() {
 		},
 	}
 
+	log := logrus.WithFields(logrus.Fields{})
+
+	// Create a new client to communicate with CAST AI API.
+	deltaHTTPClient, err := castai.NewDefaultClient(cfg.API.URL, cfg.API.Key, logger.Level, binVersion, defaultPollTimeout)
+	if err != nil {
+		log.Fatalf("new http client failed: %v", err)
+	}
+
 	client := castai.NewClient(
 		logger,
-		castai.NewDefaultClient(cfg.API.URL, cfg.API.Key, logger.Level, binVersion),
+		deltaHTTPClient,
```

Comment:
I can see it is not your change, but why here it is `logger`, whereas above it `log := logrus.WithFields(logrus.Fields{})`?

Reply:
There are two logger objects created (logger and log). Could you explain why it is needed?

Reply:
It probably was added to avoid additional fields and log levels from env when printing fatal in the main.